OpenRA.Mods.D2k/Traits/Player/BulkProductionQueue.cs

```diff
@@ -0,0 +1,219 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
+{
+	[Desc("Attach this to the player actor (not a building!) to define a new shared build queue.",
+		"Will only work together with the Production: trait on the actor that actually does the production.",
+		"You will also want to add PrimaryBuildings: to let the user choose where new units should exit.")]
+	public class BulkProductionQueueInfo : ProductionQueueInfo, Requires<TechTreeInfo>, Requires<PlayerResourcesInfo>
+	{
+		[Desc("If you build more actors of the same type,", "the same queue will get its build time lowered for every actor produced there.")]
+		public readonly bool SpeedUp = false;
+
+		[Desc("Every time another production building of the same queue is",
+			"constructed, the build times of all actors in the queue",
+			"decreased by a percentage of the original time.")]
+		public readonly int[] BuildTimeSpeedReduction = { 100, 85, 75, 65, 60, 55, 50 };
```

Comment:
I'm not sure this makes sense for a bulk production queue? I expect that the original game didn't have a production speedup here, but it would be good if somebody (@MustaphaTR?) could confirm this.

Reply:
Original game didn't have speed up on starport indeed, but i don't see a reason to not support it.

Reply:
I don't think it makes sense for the Starport to have a speedup. The `BulkProduction` can have it as an option though.

Reply:
Speedup already defaults to `false`.